tests/runtime/engine/runner.py

```diff
@@ -527,6 +527,14 @@ def print_befores_and_after_output():
             print_befores_and_after_output()
             return Runner.FAIL
 
+        if p and p.returncode == 0 and test.will_fail and not timeout:
```

Comment:
Group with above block under `if p and not timeout:` condition? Would be nice to avoid logic duplication